src/pg/datum.cpp

```diff
@@ -0,0 +1,61 @@
+#include "pgduckdb/pg/datum.hpp"
+
+extern "C" {
+#include "postgres.h"
+#include "utils/builtins.h"
+#include "utils/date.h"
+#include "utils/timestamp.h"
+#if PG_VERSION_NUM >= 160000
+#include "varatt.h"
+#endif
+}
+
+namespace pgduckdb::pg {
+int GetDetoastedDatumLen(const Datum value, bool is_bpchar) {
+	return is_bpchar ? bpchartruelen(VARDATA_ANY(value), VARSIZE_ANY_EXHDR(value))
+	                                   : VARSIZE_ANY_EXHDR(value);
+}
+const char* GetDetoastedDatumVal(const Datum value) {
+	return static_cast<const char *>(VARDATA_ANY(value));
+}
+
+bool DatumGetBool(Datum datum) {
+	return ::DatumGetBool(datum);
+}
```

Comment:
I'm kinda worried that for these tiny functions the call overhead of this could be significant. Postgres defines all these as `static inline` functions.

Reply:
To be clear, I'm thinking that maybe we should vendor in the required functions from Postgres in our own header instead.